### PR TITLE
Add basic pedigree schema

### LINKED ISSUE #838: Basic pedigree schema
See discussion in #786

A basic schema consists of:

- `sample_id` of shape `(samples, )` containing unique identifiers for samples (already implemented).
- `parent_id` of shape `(samples, parents)` containing unique identifiers of the parents of each sample where identifiers which are not found within `sample_id` are treated as unknown/missing parents.

From these we can calculate array `parent` of shape `(samples, parents)` containing the integer indices of each samples parents within the samples dimension. This should allow for faster numerical methods than using the unique identifiers directly. The justification for still including the `parent_id` array is that it is necessary to recompute the indices within the `parent` array following any alterations to the samples dimension.

Example:

```python
>>> ds = xr.Dataset()
>>> ds["sample_id"] = ["samples"], ["mom", "dad", "child"]
>>> ds["parent_id"] = ["samples", "parents"], [
...    [  ".",   "."],
...    [  ".",   "."],
...    ["mom", "dad"]
... ]
>>> # generate indices
>>> ds = parent_indices(ds)
>>> ds.parent
... array([[-1, -1],
...        [-1, -1],
...        [ 0,  1]]
>>> # subset samples
>>> ds = ds.sel(dict(samples=[True, False, True]))
>>> ds.parent
... array([[-1, -1],
...        [ 0,  1]]  # second index indicates that the child is its own parent
>>> # regenerating indices results in MergeWarning
>>> ds = parent_indices(ds)
>>> ds.parent
... array([[-1, -1],
...        [ 0, -1]]  # the child is now correctly recorded as having an unknown parent
```

### PR DESCRIPTION
**Pull request by timothymillar**
Fixes #838

### CODE REVIEW

File: sgkit/stats/pedigree.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+from sgkit import variables
+from sgkit.utils import conditional_merge_datasets, create_dataset
+
+
+def parent_indices(
+    ds: Dataset,
+    *,
+    sample_id: Hashable = variables.sample_id,
+    parent_id: Hashable = variables.parent_id,
```
Comment (jeromekelleher) [inline sgkit/stats/pedigree.py:15]: I think we should have a ``missing`` parameter which lets us specify the missing data character, and then throw an error if there's any IDs not in sample_id + missing.

How about ``missing="."``?

**Review by jeromekelleher — COMMENTED** [review]
LGTM, but I do think we need to be careful about how we treat missing data.

See suggestion above.

@tomwhite, any thoughts here on the missingness API for this?

File: sgkit/stats/pedigree.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/stats/pedigree.py:15]: This sounds reasonable to me.

Comment (timothymillar) [reply sgkit/stats/pedigree.py:15]: Good suggestion! Are others happy with `"."` as the default? I just chose it for the examples based upon similarity with the VCF spec.

Comment (jeromekelleher) [reply sgkit/stats/pedigree.py:15]: I think "." is as good as any. Another option would be "NA", but :shrug:

**Pull request merged**